mesonbuild/interpreter.py

```diff
@@ -181,12 +180,6 @@ def __init__(self):
                              'merge_from': self.merge_from_method,
                              })
 
-    def is_used(self):
-        return self.used
-
-    def mark_used(self):
```

Comment:
According to current documentation http://mesonbuild.com/Configuration.html
> You can use a single configuration_data object as many times as you like, but it becomes immutable after being passed to the configure_file function. 
So technically a copy of configuration_data should be immutable also after the source was used in configure_file. We need to reflect this change in documentation. Also this means that this PR is a functionality change and requires @jpakkane approval according to guidelines.

Reply:
Sounds reasonable, @xclaesse could you please clarify in documentation also that copy of immutable `configuration_data()` becomes mutable? Or something documenting behavior discussed above.

Reply:
Err, that is not what the above says and is in fact a good point. If you do:

    c1 = ...
    configure_file(..., configuration : c1)
    c2 = c1

Then is c2 mutable or not. It seems to me that it should remain immutable, because otherwise you can do:

    c1 = c1

It also obscures a bit the original planning behind frozenness. Originally Meson behaved like Java and Python in that variables do not contain values, but are just "references" to objects floating around in data space somewhere.

Reply:
Right, I missed the order a bit, sorry. Sent #3475 with clarification.